webdictionary/src/main/java/com/example/webdictionary/NetworkConnection.java

```diff
@@ -0,0 +1,281 @@
+        addToNet(text);
+        addToNet(oldPeersInNet);
+        //notify new peers of my old peers
+        for(String newPeerAddress: newPeersOnNet){
+            Log.d(LOG_KEY, "New Peer: " + newPeerAddress);
+            SMSPeer newPeer = new SMSPeer(newPeerAddress);
+            SMSManager.getInstance(context).sendMessage(new SMSMessage(newPeer, RequestType.AddPeers.ordinal() + " " + oldPeersInNet));
+        }
+        //notify my old peers about the new ones
+        for(SMSPeer oldPeer : oldPeersInNet){
+            Log.d(LOG_KEY, "Old Peer: " + oldPeer.getAddress());
+            SMSManager.getInstance(context).sendMessage(new SMSMessage(oldPeer, RequestType.AddPeers.ordinal() + " " + text));
+        }
+    }
+
+    //endregion
+
+    //region LeaveMethods
+    /**
+     * Sends to a given valid peer a request to leave his network
+     * @param peer The peer to send a message to.
+     * @author Edoardo Raimondi
+     */
+    public void askToLeave(SMSPeer peer){
+        if(peer == null || !peer.isValid()) throw new IllegalArgumentException();
+        SMSMessage message = new SMSMessage(peer, RequestType.LeavePermission.ordinal()+"");
+        SMSManager.getInstance(context).sendMessage(message);
+    }
+
+    /**
+     * Sends to the network a notification that a given valid SMSPeer exited the network
+     * @param peer The Peer who just left this network
+     */
+    void acceptLeave(SMSPeer peer){
+        //remove the peer
+        removeFromNet(peer.getAddress());
+        //notify my old peers about the exit
+        for(SMSPeer oldPeer : subscribers){
+            Log.d(LOG_KEY, "Old Peer: " + oldPeer.getAddress());
+            SMSManager.getInstance(context).sendMessage(new SMSMessage(oldPeer, RequestType.RemovePeers.ordinal() + " " + peer.getAddress()));
+        }
+    }
+
+    //endregion
+
+    /**
+     * Returns a space separated String with all the Peers in my Network
+     */
+    private String peersInNetwork(){
+        String netPeers = "";
+        for(SMSPeer netPeer : subscribers){
+            netPeers = netPeers.concat(netPeer.getAddress()).concat(" ");
+        }
+        return netPeers;
+    }
+
+    /**
+     * Sends to a valid peer a ping to notify that the user is still online
+     * @param peer The peer to send the ping to
+     * @throws IllegalArgumentException If the peer is null or not valid
+     */
+    public void sendPing(SMSPeer peer){
+        if(peer == null || !peer.isValid()) throw new IllegalArgumentException();
+        SMSMessage message = new SMSMessage(peer, RequestType.Ping.ordinal()+"");
+        SMSManager.getInstance(context).sendMessage(message);
+    }
+
+    /**
+     * Function called when a Ping is received from an SMSPeer,
+     * Resets the Ping timer before considering that peer offline
+     * @param peer The peer who sent the ping
+     */
+    void incomingPing(SMSPeer peer){
+        //TODO: implement this (using timers?)
+    }
+
```

Comment:
i'd have removed the todo from the review

Reply:
You're totally correct, and stuff like this should not happen. I've just implemented that method using a new class, a PingTracker object that keeps track of pings from a specific user